File: Leetcode/Medium/8-Koko-Eating-Bananas/solution.js
let minEatingSpeed = function(piles, h) {
    let [l, r] = [1, Math.max(...piles)];
    let res = r;
    while (l <= r) {
        let mid = Math.floor((r + l) / 2);
        let hours = 0;
        for (let p of piles) {
            hours += Math.ceil(p / mid);
        }
        if (hours <= 8) {
            res = Math.min(res, mid);
            r = mid  - 1
        } else {
            l = mid + 1
        }
    }
    return res;
};

File: Leetcode/Medium/8-Koko-Eating-Bananas/solution.py
import math
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l,r = 0, max(piles)
        res = r
        while l <= r:
            mid = (l+r)//2
            hours = 0
            for i in piles:
                hours += math.ceil(i/mid)
            if hours <= h:
                res = min(res,mid)
                r = mid - 1
            else:
                l = mid + 1
        return res
        